File: Interviewbit/linked-lists/remove-duplicates-from-sorted-list-ii.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
ListNode* Solution::deleteDuplicates(ListNode* A) {
    
    ListNode* tail=(ListNode*)malloc(sizeof(ListNode));
    tail->val=INT_MIN;
    tail->next=NULL;
    ListNode *p=A,*q=A,*current=tail;
    int count=0;
    while(q!=NULL){
        count=0;
        p=q;
        while(q!=NULL&&p->val==q->val){
            q=q->next;
            count++;
        }
        if(count==1){
            current->next=p;
            current=p;
        }
    }
    current->next=NULL;
    return tail->next;
}

File: Interviewbit/linked-lists/remove-duplicates-from-sorted-list-ii.java
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     public int val;
 *     public ListNode next;
 *     ListNode(int x) { val = x; next = null; }
 * }
 */
public class Solution {
    public ListNode deleteDuplicates(ListNode A) {
        
        ListNode dummy=new ListNode(Integer.MAX_VALUE);
        ListNode front=dummy,prev=A, rear=A;
        int count=0;
        while(rear!=null){
            count=0;
            while(rear!=null && prev.val==rear.val){
                rear=rear.next;
                count++;
            }
            if(count==1){
                front.next=prev;
                front=front.next;
            }
            prev=rear;
        }
        front.next=null;
        return dummy.next;
    }
}
